docs/debugging-transactions.md

```diff
@@ -0,0 +1,312 @@
+
+##### Example Code:
+
+```javascript
+// Connect to the local node using Remix's web3 provider
+const web3 = new Web3(web3.currentProvider);
+
+// Function to trace a transaction using debug_traceTransaction with opcodeLogger
+async function traceTransaction(txHash) {
+   return web3.currentProvider.send({
+      jsonrpc: '2.0',
+      id: 1,
+      method: 'debug_traceTransaction',
+      params: [txHash, { tracer: 'opcodeLogger', disableMemory: false, disableStack: false, disableStorage: false }]
+   });
+}
+
+// Example usage
+const txHash = '0x8fc90a6c3ee3001cdcbbb685b4fbe67b1fa2bec575b15b0395fea5540d0901ae';
+const result = await traceTransaction(txHash);
+console.log(result);
+```
+
+##### Response Schema:
+
+##### OpcodeTrace
+
+The `OpcodeTrace` object represents the trace of a transaction execution in the EVM.
+
+It contains the following fields:
+
+- `gas`: **number** - total gas used during the execution.
+- `failed`: **boolean** - indicates whether the transaction failed.
+- `returnValue`: **string | null** - hex-encoded return value of the transaction.
+- `structLogs`: **StructLog[]** - list of `StructLog` objects representing the opcodes executed during the transaction.
+
+##### StructLog
+
+The `StructLog` object represents an individual opcode executed during a smart contract transaction.
+
+It contains the following fields:
+
+- `pc`: **number** - program counter.
+- `op`: **string** - opcode.
+- `gas`: **number** - gas remaining.
+- `gasCost`: **number** - gas cost of the opcode.
+- `depth`: **number** - call depth.
+- `stack`: **string[]** - hex-encoded stack contents. (`null` if `disableStack` is `true`)
+- `memory`: **string[] | null** - hex-encoded memory contents. (`null` if `disableMemory` is `true`)
+- `storage`: **{ [key: string]: string } | null** - hex-encoded storage contents. (`null` if `disableStorage` is `true`)
+- `reason`: **string | null** - The reason for failure, if any. (`null` if the transaction did not fail.)
+  - In the following [format](https://besu.hyperledger.org/23.7.3/private-networks/how-to/send-transactions/revert-reason#revert-reason-format)
+
+#### Example Response:
+
+```json
+{
+   "jsonrpc": "2.0",
+   "id": 1,
+   "result": {
+      "gas": 85301,
+      "failed": true,
+      "returnValue": "",
+      "structLogs": [
+         {
+            "depth": 0,
+            "gas": 162106,
+            "gasCost": 3,
+            "memory": [],
+            "op": "PUSH1",
+            "pc": 0,
+            "reason": null,
+            "stack": [],
+            "storage": {}
+         },
+         {
+            "depth": 1,
+            "gas": 162103,
+            "gasCost": 3,
+            "memory": [],
+            "op": "PUSH1",
+            "pc": 2,
+            "reason": null,
+            "stack": [],
```

Comment:
Shouldn't the `stack` here contain the value pushed in the preceding instruction?

Reply:
Sounds good, I think with a real example, even a short one, might make it easier to understand what each value represents.

Reply:
Done, added the real opcodes response from running the `set(42)` in the `SimpleStorage` contract that's used in the example